src/ng/directive/as.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name as
+ * @restrict A
+ *
+ * @description
+ * Publishes a component controller or a DOM element into the current scope.
+ *
+ * @element ANY
+ * @param {expression} as {@link guide/expression Expression} to assign the controller.
+ *
+ * This directive allows to access to the controller of components in your
+ * current template or from current controller.
+ *
+ * @example
+ * ### Access to component controller
+ * This example demonstrates basic publishing of the component controller into the scope.
+ * <example name="asDirectiveComponentExample" module="asComponentExample">
+ *   <file name="index.html">
+ *     <script>
+ *       angular.module('asComponentExample', [])
+ *        .component('toggle', {
+ *           controller: function() {
+ *             var opened = false;
+ *             this.isOpen = function() { return opened; };
+ *             this.toggle = function() { opened = !opened; };
+ *           }
+ *         });
+ *     </script>
+ *     <toggle as="myToggle"></toggle>
+ *     <button ng-click="myToggle.toggle()">Toggle</button>
+ *     <div ng-show="myToggle.isOpen()">You are using a children component to show it.</div>
+ *   </file>
+ *   <file name="protractor.js" type="protractor">
+ *      it('should publish the toggle into the scope', function() {
+ *        var toggle = element(by.buttonText('Toggle'));
+ *        expect(toggle.evaluate('myToggle.isOpen()')).toEqual(false);
+ *        toggle.click();
+ *        expect(toggle.evaluate('myToggle.isOpen()')).toEqual(true);
+ *      });
+ *   </file>
+ * </example>
+ *
+ * This directive also allows to bind elements to the scope.
+ *
+ * ```js
+ *   var myMod = angular.module(...);
+ *   myMod.component('myPlayer', {
+ *     template: '' +
+ *       '<audio as="$ctrl.audio" src="foo.ogg"></audio>' +
+ *       '<button ng-click="$ctrl.play()">Play</button>' +
+ *       '',
+ *     controller: function() {
+ *       this.play = function() {
+ *         this.audio.play();
+ *       };
+ *     }
+ *   });
+ * ```
+ *
+ * Note that angular does not allows to access DOM elements directly
```

Comment:
This is only true when expensive checks are on, which happens on event-directives (such as `ngClick`), but not on other expressions.
Exposing DOM elements on the scope ia really bad idea and a big security concern, so we should definitely not do this.